flybywiresim/a32nx
# fix(inst): chrono & ET behavior on time shift

Fixes #[issue_no]

## Summary of Changes

Fixes elapsed time and chrono (on clock and on ND) to work with time shift by debouncing on diff more than 60s.

In simulation the elapsed time should be kept intact when changing simulated time and/or sun orientation.

## Screenshots (if necessary)



## References





## Additional context


Chrono and ET Clock use `E:ABSOLUTE TIME` to compute elapsed time, however if there is a time shift the absolute time changes with it.

The 60 seconds mark was selected as a compromise, to detect a time change. MSFS UI seems to have steps by 2min.

This is a workaround until/unless there is some other time reference that can be used to count elapsed time.


Discord username (if different from GitHub): coolll.sk

## Testing instructions


Chrono on ND and on Clock, as well as Elapsed Time, should be closer to real time.
That is:
* on unchanged time conditions
* on time shift forward
* on time shift backwards

Please note that when making a time shift/change, the chrono/timer would pause briefly until next frame.

Test Clock Chrono/CHR (on center panel, NOT on ND):
1. push CHR on clock in simulator
1. start stopwatch externally
1. time shift forward
1. time shit backwards
1. set time simulated time to original or LIVE
1. compare time with the stopwatch

Test Elapsed Time (on center panel):
1. move knob on ET to RUN in simulator
1. start stopwatch externally
1. time shift forward
1. time shit backwards
1. set time simulated time to original or LIVE
1. compare time with the stopwatch

Test Chrono on ND:
1. push Chrono above ND
1.  start stopwatch externally
1. time shift forward
1. time shit backwards
1. set time simulated time to original or LIVE
1. compare time with the stopwatch

Test for no regressions:
* do a full flight with Chrono (on ND and on CLK) and ET started
* start stopwatch externally
* compare time with the stopwatch


## How to download the PR for QA

Every new commit to this PR will cause a new A32NX artifact to be created, built, and uploaded.

1. Make sure you are signed in to GitHub
1. Click on the **Checks** tab on the PR
1. On the left side, click on the bottom **PR** tab
1. Click on the **A32NX** download link at the bottom of the page
